jena-base/src/main/java/org/apache/jena/atlas/iterator/Iter.java

```diff
@@ -29,6 +29,14 @@
 import org.apache.jena.atlas.lib.Closeable ;
 import org.apache.jena.atlas.lib.Sink ;
 
+/**
+ * Iter provides utilities for working with {@link Iterator}s.
+ *
+ * Iter should never be used as a return type or parameter type in the public contract of a class. It is only to be used
+ * inside implementation code and is instantiated only to allow method-chaining as part of a calculation.
```

Comment:
Iter itself does not impose that in its contract.  It's the rest of jena (the users of Iter) that choose to return "Iterator" not "Iter".  c.f. Map and HashMap. 

"Iter is not usually used as a return type ..." which is more advice that instruction.  If some code wishes to expose method-chaining then no problem.